File: src-tauri/src/commands/archive.rs
use crate::archive::{self as archive_impl, ArchiveContents};

#[tauri::command]
pub fn list_archive_images(archive_path: String) -> Result<Vec<String>, String> {
    archive_impl::list_images(&archive_path)
}

#[tauri::command]
pub fn get_archive_image(archive_path: String, entry_name: String) -> Result<String, String> {
    archive_impl::get_image_base64(&archive_path, &entry_name)
}

#[tauri::command]
pub fn analyze_archive_contents(archive_path: String) -> Result<ArchiveContents, String> {
    archive_impl::analyze_contents(&archive_path)
}

#[tauri::command]
pub fn extract_nested_archive(parent_path: String, nested_name: String) -> Result<String, String> {
    archive_impl::extract_nested_archive(&parent_path, &nested_name)
}

File: src/api/archive.ts
import { invoke } from "@tauri-apps/api/core";

export type ArchiveContents =
  | { type: "Images"; names: string[] }
  | { type: "NestedArchives"; names: string[] }
  | { type: "Empty" };

export async function listArchiveImages(archivePath: string): Promise<string[]> {
  return invoke<string[]>("list_archive_images", {
    archivePath,
  });
}

export async function getArchiveImage(archivePath: string, entryName: string): Promise<string> {
  return invoke<string>("get_archive_image", {
    archivePath,
    entryName,
  });
}

export async function analyzeArchiveContents(archivePath: string): Promise<ArchiveContents> {
  return invoke<ArchiveContents>("analyze_archive_contents", {
    archivePath,
  });
}

export async function extractNestedArchive(
  parentPath: string,
  nestedName: string,
): Promise<string> {
  return invoke<string>("extract_nested_archive", {
    parentPath,
    nestedName,
  });
}
